### DISCUSSION TITLE
Upgrade all non-running packages

### DISCUSSION THREAD

**Discussion by kevinholtkamp** [chocolatey/choco#2853] [Ideas]
(Sorry if this exists and I didn't find it)

There should be a way to run `choco upgrade all` but not have it upgrade running processes

**Comment by pauby** [comment]
Chocolatey doesn't upgrade the software, so has no concept of what a 'running process' is. Chocolatey upgrades the package which upgrades the software. So the work to stop / start running processes is done in the package and is the maintainers responsibility. Some packages such as Keepass and Notepad++ (I _think_) already do this.
